File: python/21_deque.py
from lib.deque import Deque

deque = Deque()

# Deque se comportando como fila comum
deque.insert_back('Antero')
deque.insert_back('Olentina')
deque.insert_back('Gaudêncio')
deque.insert_back('Hildebrando')
deque.insert_back('Iranildes')

print(deque)

removido_frente = deque.remove_front()
print(f"Removido da frente: {removido_frente}")
print(deque)

deque.insert_back('Turíbio')
print(deque)

primeiro = deque.peek_front()
print(f"Primeiro da fila: {primeiro}")
print(deque)

# USANDO RECURSOS EXCLUSIVOS DO DEQUE

# Inserção prioritário
deque.insert_front('Emerenciana')
print(deque)

# Desistência da fila
desistente = deque.remove_back()
print(f"Desistente: {desistente}")
print(deque)

# Nova inserção prioritária
deque.insert_front('Deusdete')
print(deque)

# Consultando a última pessoa da fila
ultimo = deque.peek_back()
print(f"Último: {ultimo}")
print(deque)

File: js/21_deque.js
const Deque = require('./lib/deque.js');

const deque = new Deque();

// Deque se comportando como fila comum
deque.insertBack('Antero');
deque.insertBack('Olentina');
deque.insertBack('Gaudêncio');
deque.insertBack('Hildebrando');
deque.insertBack('Iranildes');

console.log(deque.toString());

const removido_frente = deque.removeFront();
console.log(`Removido da frente: ${removido_frente}`);
console.log(deque.toString());

deque.insertBack('Turíbio');
console.log(deque.toString());

const primeiro = deque.peekFront();
console.log(`Primeiro da fila: ${primeiro}`);
console.log(deque.toString());

// USANDO RECURSOS EXCLUSIVOS DO DEQUE

// Inserção prioritário
deque.insertFront('Emerenciana');
console.log(deque.toString());

// Desistência da fila
const desistente = deque.removeBack();
console.log(`Desistente: ${desistente}`);
console.log(deque.toString());

// Nova inserção prioritária
deque.insertFront('Deusdete');
console.log(deque.toString());

// Consultando a última pessoa da fila
const ultimo = deque.peekBack();
console.log(`Último: ${ultimo}`);
console.log(deque.toString());
